Collection Form: Search for Works Asynchronously

nicktiberi opened the pull request:
Closes #1150.

Instead of searching for all works that a user has access to immediately, we now hit a JSON endpoint with an AJAX request that performs the Solr query. This limits the amount of works that are returned to a maximum of 50.

For admins, this allows them to view a subset of the works they have access to, instead of returning a gigantic dropdown list of ~10k works.

awead commented on app/services/work_zip_service.rb:
```diff
@@ -0,0 +1,45 @@
+
+    Zip::File.open(zipfile_name, Zip::File::CREATE) do |zipfile|
+      work.file_sets.each do |file_set|
+        if file_set.original_file.present? && (ability.can? :read, file_set.id)
```
You could combine the above two lines into a `select` and remove an indentation level; optional, if you think this is more readable.

cam156 commented:
@awead this PR is all fixed up.  Coveralls is angry becuase I only added unit tests for the zip service.

nicktiberi committed:
wip

nicktiberi committed:
Change collections works dropdown to populate via async search

Instead of populating the dropdown on page load with up to 10k works,
we now hit a JSON endpoint via an AJAX request. This allows admins
to search for any work they have access to, without some of them
getting left out of the original, large result set.

nicktiberi committed:
move controller to Dashboard namespace + add initial specs

rschenk committed:
Flesh out WorkSearchController spec

nicktiberi committed:
collection_form_spec changes

nicktiberi committed:
pass endpoint path into search-works controller

nicktiberi committed:
fix broken specs

nicktiberi committed:
Make every query a wildcard query so users don't have to type
whole words to get results

nicktiberi committed:
fix specs

banukutlu reviewed (approved):
lgtm 👍🏻
